recipes/jsoncpp/all/conanfile.py

```diff
@@ -70,7 +69,7 @@ def generate(self):
 
     def _patch_sources(self):
         apply_conandata_patches(self)
-        if self.settings.compiler == "Visual Studio" and self.settings.compiler.version == "11":
+        if str(self.settings.compiler) == "Visual Studio" and self.settings.compiler.version == "11":
```

Comment:
```suggestion
        if is_msvc(self) and self.settings.compiler.version in ("11", "170"):
```

A bit cheeky, but otherwise when building with Conan 2 it would not be applying the patch (`170` is still a valid version when the compiler is `msvc`).

Reply:
Love it

Reply:
This is very stupidity complicated to work around and sucks from UX :)